workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/request/client/HttpClient.java

```diff
@@ -0,0 +1,96 @@
+package com.symphony.bdk.workflow.engine.executor.request.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Generated;
+import org.apache.commons.io.IOUtils;
+import org.apache.hc.client5.http.fluent.Form;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpResponse;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+@Generated
+@SuppressWarnings("unchecked")
+@Component
+public class HttpClient {
+
+  private static final String CONTENT_TYPE = "Content-Type";
+
+  public Response execute(String method, String url, Object body, Map<String, String> headers)
+      throws IOException {
+    HttpResponse httpResponse;
+    Request request;
+    switch (method) {
+      case "GET":
+        request = Request.get(url);
+        httpResponse = this.executeWithoutBody(request);
+        break;
+      case "POST":
+        request = Request.post(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "PUT":
+        request = Request.put(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "DELETE":
+        request = Request.delete(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "PATCH":
+        request = Request.patch(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "HEAD":
+        request = Request.head(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      case "OPTIONS":
+        request = Request.options(url);
+        httpResponse = this.executeWithBody(request, body, headers);
+        break;
+      default:
+        throw new RuntimeException("Method is not supported");
+    }
+
+    ClassicHttpResponse classicHttpResponse = (ClassicHttpResponse) httpResponse;
+    int responseCode = httpResponse.getCode();
+
+    if (classicHttpResponse.getEntity() != null && classicHttpResponse.getEntity().getContent() != null) {
+      return new Response(responseCode,
+          IOUtils.toString(classicHttpResponse.getEntity().getContent(), StandardCharsets.UTF_8));
+    } else {
+      return new Response(responseCode, "");
+    }
+  }
+
+  private HttpResponse executeWithoutBody(Request request) throws IOException {
+    return request.execute().returnResponse();
+  }
+
+  private HttpResponse executeWithBody(Request request, Object body, Map<String, String> headers)
+      throws IOException {
+
+    // set body
+    String contentType = headers.get(CONTENT_TYPE);
+    if (ContentType.MULTIPART_FORM_DATA.getMimeType().equals(contentType)) {
```

Comment:
can't we have the body as an object in the model and then if the content type is form, enforce it to be a map directly? instead of passing a JSON map inside YAML

Reply:
To be able to enforce cast to a map, the body should not be a String. So in case of multipart-form-data, if we provide the body as an object it's fine. 
However, this doesn't work for other content types where we need to provide the body in a given format (in JSON for application/json, and a different format for text/plain). So to write it in the required format, we need a String. 
In next commit, I'm changing json-schema to define the body as string and object, and it is object in the java model. Like this we can support both: when we use multipart type, we provide an object, and a string for other types. This can be ambiguous though.